roles/ceph-update/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: check if osds are up
+  shell: ceph osd stat | awk '{print $5}' | egrep -sq "$(ceph osd stat | awk '{print $3}')"
+  register: result
+  until: result.rc == 0
+  retries: 10
+  delay: 10
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+
+- name: restart ceph mons
+  service: name=ceph-mon-all state=restarted
+
+# Setting noout to prevent cluster from rebalancing after service restart
+- name: set osd noout
+  command: ceph osd set noout
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: inventory_hostname in groups['ceph_osds'][0]
+
+# Setting osd values for first osd to reduce performance degradation of client I/O
+- name: set osd noscrub
+  command: ceph osd set noscrub
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: inventory_hostname in groups['ceph_osds'][0]
+
+- name: set osd nodeep scrub
+  command: ceph osd set nodeep-scrub
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: inventory_hostname in groups['ceph_osds'][0]
+
+- name: restart ceph osds
+  service: name=ceph-osd-all state=restarted
+
+# Unset the osd values only after ceph on last osd node has restarted
+- name: unset osd noout
+  command: ceph osd unset noout
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: inventory_hostname in groups['ceph_osds'][{{ groups['ceph_osds']|length - 1 }}]
```

Comment:
`when: inventory_hostname == groups['ceph_osds'][-1]`

That will match the current host name to the last entry in the ceph_osds group, which I think is what you're trying to achieve.